### ISSUE TITLE
Write tests to verify pre and post upgrade state for event bus

### ISSUE THREAD

**Issue by abbi-gaurav** [kyma-project/kyma#2600]
**Description**

To ensure smooth upgradability with each release, we need to have tests for the event bus that ensures a state reached in a pre-upgrade version is kept and the business flow stays uninterrupted post upgrade.

**What to verify**
- Create Kyma-susbcription with related resources in `pre-upgrade`
- Verify subscriber can receive the event `pre-upgrade`
- Verify subscriber can receive the event `post-upgrade`

Refer 
- https://github.com/kyma-project/kyma/tree/master/tests/end-to-end/upgrade

**AC**
- [x] There are upgrade tests for event-bus.

**Comment by radufa** [comment]
the post upgrade test should also check somehow that each Event-Bus component (the image) was updated (perhaps checking the version number of each component, if it's possible)

**Comment by abbi-gaurav** [comment]
Putting it on hold until the sample or way to write tests is finalized.

**Comment by montaro** [comment]
Blocked by https://github.com/kyma-project/kyma/issues/3017

**Comment by abbi-gaurav** [comment]
As discussed during the review, it would be nice to have a different payload in the post-upgrade scenario. This will help us avoid the pitfalls when for unknown reasons a messaging middleware could redeliver the pre-upgrade event and the test will assume everything is working fine.
